### PR TITLE
ci: add ignore to known deadlink

### PR DESCRIPTION
**Pull request by kazuk**
## Issue number and link

Fixes: #166

add "ignore known deadlink"

## Describe your changes

- add `deadlink.expected` file
- sort/uniq diff to `cargo-deadlinks` result

## Checklist before requesting a review

- [x] I follow the [Semantic Pull Requests](https://github.com/zeke/semantic-pull-requests) rules (bugfix/feature)
- [x] I specified links to related issues (must: bugfix, want: feature)
- [x] I have performed a self-review of my code (bugfix/feature)
- [ ] I have added thorough tests (bugfix/feature)
- [x] I have edited `## [Unreleased]` section in `CHANGELOG.md` following [keep a changelog](https://keepachangelog.com/en/1.0.0/) syntax (bugfix/feature)
- [ ] I {made/will make} a related pull request for [documentation repo](https://github.com/SpringQL/SpringQL.github.io) (feature)

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -102,8 +102,17 @@ cargo d
 [tasks.deadlink]
 script = ['''
 #!/usr/bin/env bash -eux
-cargo deadlinks --check-http --ignore-fragments
+
+mkdir -p target
+set +e
+cargo deadlinks --check-http --ignore-fragments > target/cargo-deadlinks.out
```
Comment (laysakura) [inline Makefile.toml:108]: Want to check stdout in CI log.

```suggestion
cargo deadlinks --check-http --ignore-fragments | tee target/cargo-deadlinks.out
```

File: Makefile.toml [hunk 2]
```diff
@@ -102,8 +102,17 @@ cargo d
...
+set -e
+cat target/cargo-deadlinks.out | sort | uniq > target/cargo-deadlinks.sort
```
Comment (laysakura) [inline Makefile.toml:110]: Add some comments like:

```bash
# Create a deadlink log file that includes known deadlink messages
```

**Review by laysakura — CHANGES_REQUESTED** [review]

File: Makefile.toml [hunk 3]
```diff
@@ -102,8 +102,27 @@ cargo d
...
+# check diff for `deadlinks.expected`
+cat deadlinks.expected | sort | uniq > target/cargo-deadlinks.expected.sort
+diff -u target/cargo-deadlinks.expected.sort target/cargo-deadlinks.sort
+
 mlc --ignore-path target
+
+''']
+
+[tasks.deadlink-expect]
```
Comment (laysakura) [inline Makefile.toml:120]: want to depend on this task from `deadlink` task?

**Review by laysakura — CHANGES_REQUESTED** [review]

File: Makefile.toml (hunk 3 above)
Comment (kazuk) [reply Makefile.toml:120]: No, this task seeds output expectation to `deadlink.expected`

Comment (laysakura) [reply Makefile.toml:120]: Ah, sorry. Got it.

**Review by laysakura — APPROVED** [review]

**Pull request merged**